code/STLLoader.cpp

```diff
@@ -449,26 +449,29 @@ bool STLImporter::LoadBinaryFile()
     aiVector3D *vp = pMesh->mVertices = new aiVector3D[pMesh->mNumVertices];
     aiVector3D *vn = pMesh->mNormals = new aiVector3D[pMesh->mNumVertices];
 
+    typedef aiVector3t<float> aiVector3F;
+    aiVector3F* theVec;
+    
     for ( unsigned int i = 0; i < pMesh->mNumFaces; ++i ) {
         // NOTE: Blender sometimes writes empty normals ... this is not
         // our fault ... the RemoveInvalidData helper step should fix that
-        ::memcpy( vn, sz, sizeof( aiVector3D ) );
-        sz += sizeof(aiVector3D);
-        *(vn+1) = *vn;
-        *(vn+2) = *vn;
-        vn += 3;
-
-        ::memcpy( vp, sz, sizeof( aiVector3D ) );
-        ++vp;
-        sz += sizeof(aiVector3D);
-
-        ::memcpy( vp, sz, sizeof( aiVector3D ) );
-        ++vp;
-        sz += sizeof(aiVector3D);
-
-        ::memcpy( vp, sz, sizeof( aiVector3D ) );
-        ++vp;
-        sz += sizeof(aiVector3D);
+
+        // There's one normal for the face in the STL; use it three times
```

Comment:
This will not work when the memory is not aligned in the right way. That was the reason to remove the c-like casts.